src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -80,19 +81,23 @@ private static void dumpMetadata(Checker checker, String path) throws Checkstyle
             validFiles.add(new File(path));
         }
         else {
-            final List<String> moduleFolders = Arrays.asList("checks", "filters", "filefilters");
-            for (String folder : moduleFolders) {
-                try (Stream<Path> files = Files.walk(Paths.get(path
-                        + "/" + folder))) {
-                    validFiles.addAll(
-                            files.map(Path::toFile)
-                            .filter(file -> {
-                                return file.getName().endsWith("SuppressWarningsHolder.java")
-                                        || file.getName().endsWith("Check.java")
-                                        || file.getName().endsWith("Filter.java");
-                            })
-                            .collect(Collectors.toList()));
+            try {
+                final List<String> moduleFolders = Arrays.asList("checks", "filters", "filefilters");
+                for (String folder : moduleFolders) {
+                    try (Stream<Path> files = Files.walk(Paths.get(path
+                            + "/" + folder))) {
+                        validFiles.addAll(
+                                files.map(Path::toFile)
+                                        .filter(file -> {
+                                            return file.getName().endsWith("SuppressWarningsHolder.java")
+                                                    || file.getName().endsWith("Check.java")
+                                                    || file.getName().endsWith("Filter.java");
+                                        })
+                                        .collect(Collectors.toList()));
+                    }
                 }
+            }catch (FileNotFoundException e){
+                System.out.println("Folders not present");
```

Comment:
Which folder? This information is present in the exception. Why not share it with the user?

Reply:
@pbludov  Thanks sir I will check again

Reply:
Sir how to share this with the user

Reply:
> Sir how to share this with the user

Like this (for example):
```java
  System.out.println("Folder not present: " + exception.getFile());
```
Note that `Files.walk` throws `NoSuchFileException`, not `FileNotFoundException`.